src/dodal/devices/xspress3/xspress3.py

```diff
@@ -0,0 +1,146 @@
+from enum import Enum
+
+from numpy import float64
+from numpy.typing import NDArray
+from ophyd_async.core import (
+    AsyncStatus,
+    Device,
+    DeviceVector,
+    set_and_wait_for_value,
+    wait_for_value,
+)
+from ophyd_async.epics.signal.signal import (
+    epics_signal_r,
+    epics_signal_rw,
+    epics_signal_rw_rbv,
+)
+
+from dodal.devices.xspress3.xspress3_channel import (
+    AcquireState,
+    Xspress3Channel,
+    Xspress3ROIChannel,
+)
+from dodal.log import LOGGER
+
+
+class TriggerMode(str, Enum):
+    SOFTWARE = "Software"
+    HARDWARE = "Hardware"
+    BURST = "Burst"
+    TTL_Veto_Only = "TTL Veto Only"
+    IDC = "IDC"
+    SOTWARE_START_STOP = "Software Start/Stop"
+    TTL_BOTH = "TTL Both"
+    LVDS_VETO_ONLY = "LVDS Veto Only"
+    LVDS_both = "LVDS Both"
+
+
+class UpdateRBV(str, Enum):
+    DISABLED = "Disabled"
+    ENABLED = "Enabled"
+
+
+'''
+class EraseState(str, Enum):
+    """
+    Epic does not actually have a str value so the only way to set it is
+    to pass int(position) but we still need an Enum to bypass checking in signal
+    """
+
+    DONE = ""
+    ERASE = ""
+
+'''
+
+
+class AcquireRBVState(str, Enum):
+    DONE = "Done"
+    ACQUIRE = "Acquiring"
+
+
+class DetectorState(str, Enum):
+    IDLE = "Idle"
+    ACQUIRE = "Acquire"
+    READOUT = "Readout"
+    CORRECT = "Correct"
+    Saving = "Saving"
+    ABORTING = "Aborting"
+    ERROR = "Error"
+    WAITING = "Waiting"
+    INTILTIALIZING = "Initializing"
+    DISCONNECTED = "Disconnected"
+    ABORTED = "Aborted"
+
+
+class Xspress3(Device):
+    """Xpress/XpressMini is a region of interest (ROI) picker that sums the detector
+    output into a scaler with user-defined regions. It is often used as a signal
+     discriminator to provide better energy resolution in X-ray detection experiments.
+    """
+
+    def __init__(
+        self, prefix: str, name: str = "", num_channels: int = 1, timeout=1
+    ) -> None:
+        self.channels = DeviceVector(
+            {i: Xspress3Channel(f"{prefix}C{i}_") for i in range(1, num_channels + 1)}
+        )
+        """MCA on/off switch readback"""
+        self.get_roi_calc_status = DeviceVector(
+            {
+                i: epics_signal_rw(float, f"{prefix}MCA{i}:Enable_RBV")
+                for i in range(1, num_channels + 1)
+            }
+        )
+        """start and size of the MCA array"""
```

Comment:
what is MCA array specifically?

Reply:
multi-channel analyser, unlike the single channel analyser(SCA) it return an array of SCA counts over a user defined time/counter/size.   So in term of xpress3, rather than arming/staging it every time you want 1 data point, you arm/stage it for x number of point and it return an array instead of a scaler.